### PR TITLE
Limit stream spans gRPC message to ~4MiB

### PR DESCRIPTION
**Pull request by guilload**
### Description
Per title.

### How was this PR tested?
Tested in Jaeger UI

### CODE REVIEW

File: quickwit/quickwit-jaeger/src/lib.rs [hunk 1]
```diff
@@ -337,32 +337,36 @@ impl JaegerService {
...
-                if chunk.len() == CHUNK_SIZE {
+                if chunk.len() == MAX_CHUNK_LEN
+                    || chunk_num_bytes + span_num_bytes > MAX_CHUNK_NUM_BYTES
+                {
                     num_spans_total += chunk.len();
                     num_bytes_total += chunk_num_bytes;
 
-                    let chunk_size = spans.len().min(CHUNK_SIZE);
-                    let chunk = mem::replace(&mut chunk, Vec::with_capacity(chunk_size));
+                    let chunk_len = MAX_CHUNK_LEN.min(spans.len() + 1);
```
Comment (fmassot) [inline quickwit/quickwit-jaeger/src/lib.rs:358]: why +1?

Comment (guilload) [reply quickwit/quickwit-jaeger/src/lib.rs:358]: To account for the span that we popped from `spans` but have not appended yet to `chunk`. Will add a comment.

File: quickwit/quickwit-jaeger/src/lib.rs [hunk 2]
```diff
@@ -337,32 +337,36 @@ impl JaegerService {
...
+                    let chunk = mem::replace(&mut chunk, Vec::with_capacity(chunk_len));
                     if let Err(send_error) = tx.send(Ok(SpansResponseChunk { spans: chunk })).await
                     {
                         debug!("Client disconnected: {send_error:?}");
                         return;
                     }
-                    record_send(operation_name, CHUNK_SIZE, chunk_num_bytes);
+                    record_send(operation_name, MAX_CHUNK_LEN, chunk_num_bytes);
```
Comment (fmassot) [inline quickwit/quickwit-jaeger/src/lib.rs:365]: are we sure that we record always `MAX_CHUNK_LEN`? Why not putting `chunk.len()`?

**Review by fmassot — APPROVED** [review]
One or two  comments would have helped me.

**Pull request merged**